pages/signin.js

```diff
@@ -16,36 +16,41 @@ import SignInOrJoinFree from '../components/SignInOrJoinFree';
 import { withUser } from '../components/UserProvider';
 
 class SigninPage extends React.Component {
-  static getInitialProps({ query: { token, next, form } }) {
+  static getInitialProps({ query: { token, next, form, tfa } }) {
     // Decode next URL if URI encoded
     if (next && next.startsWith('%2F')) {
       next = decodeURIComponent(next);
     }
 
     next = next && isValidRelativeUrl(next) ? next : null;
-    return { token, next, form: form || 'signin' };
+    return { token, next, form: form || 'signin', tfa };
   }
 
   static propTypes = {
     form: PropTypes.oneOf(['signin', 'create-account']).isRequired,
     token: PropTypes.string,
     next: PropTypes.string,
+    tfa: PropTypes.string,
```

Comment:
I understand that `2fa` might be invalid and this is why you used `tfa`, I recommend a longer and more explicit name.